1.9 release notes (#1027)
cert-manager/website: content/next-docs/release-notes/release-notes-1.9.md

---
title: Release 1.9
description: 'cert-manager release notes: cert-manager v1.9'
---

## v1.9.0

cert-manager 1.9 adds alpha support for using cert-manager `Certificate`s in scenarios where the ordering of the Relative Distinguished Names sequence that constitutes an X.509 certificate's subject needs to be preserved, improves ability to configure the `Certificate` created via ingress-shim using annotations on the `Ingress` resource and introduces various changes/improvements in contributor flow, most importantly defines a new make based contributor workflow.

> 

Reviewer: This blank line seems accidental

---
title: Release 1.9
description: 'cert-manager release notes: cert-manager v1.9'
---

## v1.9.0

> cert-manager 1.9 adds alpha support for using cert-manager `Certificate`s in scenarios where the ordering of the RDN sequence that constitutes an X.509 certificate's subject needs to be preserved, improves ability to configure the `Certificate` created via ingress-shim using annotations on the `Ingress` resource and introduces various changes/improvements in contributor flow, most importantly defines a new make based contributor workflow.

Reviewer: There's no apostrophe in:
> `Certificate`s

But there is one below...

### Major Themes

#### `LiteralCertificateSubject` feature

> cert-manager's `Certificate` allows to configure the subject fields of the X.509 certificate via `spec.subject` and `spec.commonName` fields. The [X.509 spec](https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.6) states that the subject is an (ordered) sequence of Relative Distinguished Names (RDN). cert-manager does not strictly abide by the spec when encoding the subject fields from `Certificate`'s spec, for example, the order of the RDN sequence may not be preserved.

Reviewer: RDN is used on line 8 and expanded on line 14... My natural inclination is to put the expansion on 8, or at least a forward link to a thing...

> For vast majority of the users this does not matter, but there are specific cases that require to be able to define the exact ordered RDN sequence, for example, if the certificate is used for LDAP authentication and the RDN sequence represents a [location in LDAP directory tree](https://ldapwiki.com/wiki/Directory%20Information%20Tree), see [`cert-manager#3203`](https://github.com/cert-manager/cert-manager/issues/3203).

Reviewer: ```suggestion
For the vast majority of the users this does not matter, but there are specific cases that require defining the exact ordered RDN sequence, for example, if the certificate is used for LDAP authentication and the RDN sequence represents a [location in LDAP directory tree](https://ldapwiki.com/wiki/Directory%20Information%20Tree). See [`cert-manager#3203`](https://github.com/cert-manager/cert-manager/issues/3203)
```

> For these use cases, a new alpha `LiteralSubject` field has been added to `Certificate`'s spec where users can pass an RDN sequence, for example:

Reviewer: There's one here in:
> `Certificate`'s spec

I understand there's technically a distinction, and would suggest this:

```suggestion
For these use cases, a new alpha `LiteralSubject` field has been added to the `Certificate` spec where users can pass an RDN sequence, for example:
```

```
apiVersion: cert-manager.io/v1
kind: Certificate
metadata:
  name: test
spec:
  secretName: test
  literalSubject: "C=US,O=myOrg,CN=someName"
...
To use this field, the alpha `LiteralCertificateSubject` feature gate needs to be enabled both on cert-manager [controller](../cli/controller.md) and [webhook](../cli/webhook/).
`spec.literalSubject` is mutually exclusive with `spec.commonName` and `spec.subject`.
```

> This feature is aimed at the specific scenarios where an exact RDN sequence needs to be defined. We do not intend to deprecate the existing `spec.subject` and `spec.commonName` fields and we recommend that folks keep using those fields in all other cases as they have better validation and visibility.

Reviewer: ```suggestion
This feature is aimed at the specific scenario where an exact RDN sequence needs to be defined. We do not intend to deprecate the existing `spec.subject` and `spec.commonName` fields and we recommend that folks keep using those fields in all other cases as they have better validation and visibility.
```

#### ingress-shim `Certificate`'s configuration

cert-manager 1.9 adds the ability to configure `Certificate`'s `spec.revisionHistoryLimit` and `spec.privateKey` for [ingress-shim](../usage/ingress/) `Certificate`s via [annotations on the `Ingress` resource](../usage/ingress/#supported-annotations).

> This should allow folks to configure ingress-shim `Certificate`s according to best practices (i.e by setting `Certificate`'s `spec.privateKey.rotationPolicy` to `Always`). In the future we would like to design a better mechanism to  configure these `Certificate`s.

Reviewer: ```suggestion
This should allow folks to configure ingress-shim `Certificate`s according to best practices (i.e by setting `Certificate`'s `spec.privateKey.rotationPolicy` to `Always`). In the future we would like to design a better mechanism to configure these `Certificate`s.
```

> We advise caution when using `Ingress` annotations as there is no validation of the annotations at the `Ingress` creation time.

Reviewer: ```suggestion
We advise caution when using `Ingress` annotations as there is no validation of the annotations at `Ingress` creation time.
```

#### contribution workflow

Over the past couple of months there have been a number of discussions in regards to contributor experience and project health, partially triggered by the awesome community discussions in cert-manager's KubeCon booth and also by the work done to move cert-manager to CNCF's incubating stage.
For example, we've [clarified our feature policy](../contributing/policy.md) and discussed the process of building cert-manager's roadmap.

##### make workflow

> cert-manager 1.8 introduced a new make based workflow alongside the existing Bazel workflow. The work to improve the make workflow was continued in 1.9 and our [contributor flow](../contributing/building.md) has been redefined to use make commands. This should make building and testing cert-manager easier with faster build and test times, easier debugging and less complexity.

Reviewer: ```suggestion
cert-manager 1.8 introduced a new `make`-based workflow alongside the existing Bazel workflow. The work to improve the `make` workflow was continued in 1.9 and our [contributor flow](../contributing/building.md) has been redefined to use `make` commands. This should make building and testing cert-manager easier with faster build and test times, easier debugging and less complexity.
```
